### PR TITLE
[Java] : Add code to detect anagrams 

### PR DESCRIPTION
**Pull request by Ritabrata1080**
- [x] File header updated
- [x] No new warnings added by PR
- [x] No commented code present in PR

### CODE REVIEW

**Review by tanweerkamal425 — APPROVED** [review]

**Review by AGNIVA029 — APPROVED** [review]
approved

File: src/practice/LC_2.java [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+ * You are given two non-empty linked lists representing two non-negative integers. The digits are stored in reverse order, and each of their nodes contains a single digit. Add the two numbers and return the sum as a linked list.
+ * <p>
+ * You may assume the two numbers do not contain any leading zero, except the number 0 itself.
+ */
 
-    You may assume the two numbers do not contain any leading zero, except the number 0 itself.
-  */
-  
- class Solution {
+class LC_2 {
```
[the remark this answers is not in the document]
Comment (Ritabrata1080) [reply src/practice/LC_2.java:15]: LC_2 is a convention we are following here. 
Leetcode_{problemNumber}. This naming convention is kept to simplify. Whenever we will look into the class name, we can get idea about the problem number and which platform.

File: src/practice/LC_242_Valid_Anagram.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+package practice;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class LC_242_Valid_Anagram {
```
[the remark this answers is not in the document]
Comment (Ritabrata1080) [reply src/practice/LC_242_Valid_Anagram.java:21]: Addressed.

File: src/practice/LC_242_Valid_Anagram.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    public static boolean isAnagram_approach2(String s, String t) {
+        if (s.length() != t.length()) {
+            return false;
+        }
+        char[] c1 = s.toCharArray();
+        char[] c2 = t.toCharArray();
+        Arrays.sort(c1);
+        Arrays.sort(c2);
+        int i = 0;
+        int j = 0;
```
[the remark this answers is not in the document]
Comment (Ritabrata1080) [reply src/practice/LC_242_Valid_Anagram.java:58]: Defining variables on same line gives code smell caught by sonar lint.

**Review by tanweerkamal425 — APPROVED** [review]

**Review by AGNIVA029 — APPROVED** [review]

**Pull request merged**